postgrest_py/_async/request_builder.py

```diff
@@ -36,10 +38,13 @@ async def execute(self) -> Tuple[Any, Optional[int]]:
             self.path,
             json=self.json,
         )
-        return process_response(self.session, r)
+        try:
+            return APIResponse.from_http_request_response(r)
```

Comment:
The returning type hint of the `execute` method should be updated to `APIResponse`